### ISSUE TITLE
no-unused-vars false positive in for-in loops

### ISSUE THREAD

**Issue by lencioni** [eslint/eslint#13163]
**Tell us about your environment**

Node version: v10.16.0
npm version: v6.14.2
Local ESLint version: v6.8.0 (Currently used)
Global ESLint version: Not found

This appeared during an update from eslint 5.16.0 to 6.8.0.

**What parser (default, Babel-ESLint, etc.) are you using?** babel-eslint

Config, probably something like this:

```json
{
  "rules": {
    "no-unused-vars": ["error", { "vars": "all", "args": "after-used", "ignoreRestSiblings": true }],
  }
}
```

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const obj = { foo: true };
for (const k in obj) {
  if (Object.prototype.hasOwnProperty.call(obj, k)) {
    console.log(k);
  }
}
```

```bash
npx eslint foo.js
```

**What did you expect to happen?**
No errors

**What actually happened? Please include the actual, raw output from ESLint.**
```
error    'k' is defined but never used 
```

**Are you willing to submit a pull request to fix this bug?**
Sure, if I can find the time

**Comment by lencioni** [comment]
Seems to also affect for-of loops

**Comment by mdjermanovic** [comment]
Hi @lencioni, thanks for the bug report!

This is probably same as babel/babel-eslint#791 issue, can you please try with the latest `babel-eslint` version?

**Comment by lencioni** [comment]
Good call! Bumping babel-eslint 11.0.0-beta.0 to 11.0.0-beta.2 seems to do the trick! Thank you!
